### PR TITLE
C patches to DMI DPI

### PR DESCRIPTION
**Pull request by dpetrisko**
Hi, I'm experimenting with some elements of the DMI DPI debugger system. When I was compiling with VCS, I encountered some errors that seem to be related to C/C++ differences. This patches fixes them as non-invasively as possible

Signed-off-by: Dan Petrisko <petrisko@cs.washington.edu>

### CODE REVIEW

File: hw/dv/dpi/common/tcp_server/tcp_server.c [hunk 1]
```diff
@@ -21,7 +21,11 @@
 /**
  * Simple buffer for passing data between TCP sockets and DPI modules
  */
+#ifdef __cplusplus
```
Comment (timothytrippel) [inline hw/dv/dpi/common/tcp_server/tcp_server.c:24]: Do you need this `ifdef`? can you not just make this a:
`static const int BUFSIZE_BYTE = 256;`?

File: hw/dv/dpi/dmidpi/dmidpi.c [hunk 2]
```diff
@@ -31,7 +31,11 @@ const int IDCODEVAL = 0x04F5484D;
 // [3:0]    0x1  - Protocol version (0.13)
 const int DTMCSRVAL = 0x00000071;
 
+#ifndef __cplusplus
```
Comment (timothytrippel) [inline hw/dv/dpi/dmidpi/dmidpi.c:34]: same here, do we need the underlying type for the enum? or can you just drop the `ifdef`s and the `: uint8_t` on these enums and everything still works?

**Review by timothytrippel — APPROVED** [review]
Thanks for the fixes @dpetrisko ! I left a couple suggestions otherwise LGTM. Also, I think you need to hit this with clang-format to get it to pass CI. To ensure you run the right clang-format version if you run `./bazelisk.sh run //:clang_format_fix` Bazel should trigger the format for you.

File: hw/dv/dpi/common/tcp_server/tcp_server.c (hunk 1 above)
Comment (dpetrisko) [reply hw/dv/dpi/common/tcp_server/tcp_server.c]: static const int BUFSIZE_BYTE = 256 results in:
```
opentitan/hw/dv/dpi/common/tcp_server/tcp_server.c:29:8: error: variably modified 'buf' at file scope
   29 |   char buf[BUFSIZE_BYTE];
```
I could drop the ifdef if we do
```
#define BUFSIZE_BYTE 256
```

I'm fine without the ifdef if we #define it. Is there was a C++ reason to prefer static const int over #define INT ?

File: hw/dv/dpi/dmidpi/dmidpi.c (hunk 2 above)
Comment (dpetrisko) [reply hw/dv/dpi/dmidpi/dmidpi.c]: Can confirm works with my use case in C, will let CI tell the full story :)

File: hw/dv/dpi/common/tcp_server/tcp_server.c (hunk 1 above)
Comment (timothytrippel) [reply hw/dv/dpi/common/tcp_server/tcp_server.c]: ah right this is C not C++.  the #define is good in that case.

The reason comes from our [style guide](https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros), but its ok in this case b/c of the given circumstances (compiling with both C and C++).

**Pull request merged**